src/PostgREST/Request/ApiRequest.hs

```diff
@@ -230,12 +238,11 @@ userApiRequest conf@AppConfig{..} dbStructure req reqBody
                       ((<> ".") <$> "not":M.keys operators) ++
                       ((<> "(") <$> M.keys ftsOperators)
   isEmbedPath = T.isInfixOf "."
-  isTargetingProc = case target of
-    TargetProc _ _ -> True
-    _              -> False
-  isTargetingDefaultSpec = case target of
-    TargetDefaultSpec _ -> True
-    _                   -> False
+  (isTargetingProc, targetProcIsRootSpec, isTargetingDefaultSpec) = case path of
+    [] | isJust configDbRootSpec -> (True, True, False)
+       | otherwise               -> (False, False, True)
+    ["rpc", _]                   -> (True, False, False)
+    _                            -> (False, False, False)
```

Comment:
Alright, the procedure error handling https://github.com/PostgREST/postgrest/pull/1841#discussion_r633287931 without the fallback procedure https://github.com/PostgREST/postgrest/pull/1841#discussion_r626902462 looks better now. There are some changes that needed to be made, like replacing the `target` function call in some places in order to make it the last error to be checked (in these selected lines for instance).

Need some feedback and suggestions to make this change a bit cleaner.

Reply:
This will change once you change the `findProc` to return an Either; let's see if it's simpler then. For now, from the previous version, it looks like `isTargetingProc/isTargetingDefaultSpec` could be outside here and go next to `ApiRequest.Target`; and become independent functions.

Reply:
Modified the `findProc` to return an `Either ApiRequestError Target` but couldn't see how to refactor this part of the code to make it look cleaner without complicating things after this change. Mostly because these functions are needed before any `Target` is assigned. I'll leave it like this for now and open the PR for reviews.